go/libraries/doltcore/sqle/table_editor.go

```diff
@@ -41,46 +42,39 @@ var ErrDuplicatePrimaryKeyFmt = "duplicate primary key given: (%v)"
 // editor after every SQL statement is incorrect and will return incorrect results. The single reliable exception is an
 // unbroken chain of INSERT statements, where we have taken pains to batch writes to speed things up.
 type tableEditor struct {
-	t        *WritableDoltTable
-	tableEd  *subTableEditor
-	indexEds []*subTableEditor
-}
-
-type subTableEditor struct {
-	parent       *tableEditor
-	sch          schema.Schema
+	t            *WritableDoltTable
 	ed           *types.MapEditor
-	idx          schema.Index
 	insertedKeys map[hash.Hash]types.Value
 	addedKeys    map[hash.Hash]types.Value
 	removedKeys  map[hash.Hash]types.Value
+	affectedKeys map[hash.Hash]types.Value
+	indexEds     []*indexEditor
+}
+
+type indexEditor struct {
```

Comment:
I like this abstraction a lot, much cleaner. Just make sure that it's thoroughly tested, this code has a lot of weird edge cases. It looks to me like you might have a bug on delete?